tests/st/calico_upgrade/test_calico_upgrade.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2015-2017 Tigera, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import random
+
+import json
+import yaml
+from nose_parameterized import parameterized
+
+from tests.st.utils.utils import calicoctl, calicoctlv2, \
+    name, set_version_etcdv2, dump_etcdv2, dump_etcdv3, \
+    set_ready_etcdv2, get_value_etcdv2, get_value_etcdv3, \
+    calicoupgrade, wipe_etcdv2, wipe_etcdv3, get_ip, clean_calico_data
+from tests.st.utils.v1_data import data
+
+ETCD_SCHEME = os.environ.get("ETCD_SCHEME", "http")
+ETCD_CA = os.environ.get("ETCD_CA_CERT_FILE", "")
+ETCD_CERT = os.environ.get("ETCD_CERT_FILE", "")
+ETCD_KEY = os.environ.get("ETCD_KEY_FILE", "")
+ETCD_HOSTNAME_SSL = "etcd-authority-ssl"
+
+logging.basicConfig(level=logging.DEBUG, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+tests = [
+    ("bgppeer_long_node_name", False),
+    ("bgppeer_dotted_asn", False),
+    ("hep_bad_label", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+    ("hep_tame", False),
+    ("hep_mixed_ip", False),
+    ("hep_label_too_long", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+    ("hep_long_fields", False),
+    ("hep_name_too_long", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+    ("ippool_mixed", False),
+    ("ippool_v4_small", False),
+    ("ippool_v4_large", False),
+    ("node_long_name", False),
+    ("node_tame", False),
+    ("policy_long_name", False),
+    ("policy_big", False),
+    ("policy_tame", False),
+    ("wep_bad_workload_id", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+    ("profile_big", False),
+    ("profile_tame", False),
+    ("wep_lots_ips", False),
+    ("wep_similar_name", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+    ("wep_similar_name_2", False),
+    ("do_not_track", False),
+    ("prednat_policy", False),
+    ("profile_long_labels", True,
+     "ERROR: Error converting data, check output for details and resolve issues"),
+]
+random.shuffle(tests)
+
+
+@parameterized.expand([
+    "dry-run",
+    "start",
+])
+def test_dry_run(cmd):
```

Comment:
I think this is badly named.  It's testing an empty datastore rather than the dry-run command

Reply:
Addressed: 
def test_empty_datastore(cmd):